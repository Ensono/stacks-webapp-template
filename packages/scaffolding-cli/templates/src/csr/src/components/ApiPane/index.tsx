import {Typography, CircularProgress} from "@material-ui/core"
import React, {useEffect} from "react"
import {DispatchToProps, MenuState} from "../../interfaces/sagas.interface"
import {Container} from "./components"
import RestaurantListComponent from "../RestaurantList"

// type ApiPaneProps = {
//     menuItems: RestaurantItemProps[]
//     isLoading: boolean
// }

type ApiPaneProps = MenuState & DispatchToProps

const ApiPane: React.FC<ApiPaneProps> = ({
    loading,
    errors,
    data,
    fetchPosts,
}: ApiPaneProps) => {
    useEffect(() => {
        fetchPosts()
    }, [fetchPosts])
    return (
        <Container item>
            <Typography variant="h1">Latest menus:</Typography>
            {loading && <CircularProgress size={26} />}
            {!loading && errors && (
                <Typography variant="h2">{errors}</Typography>
            )}
            {!loading && data?.length === 0 && (
                <Typography variant="h2">No results</Typography>
            )}
            {!loading && data?.length >= 1 && (
                <RestaurantListComponent restaurantList={data} />
            )}
        </Container>
    )
}

export default ApiPane
