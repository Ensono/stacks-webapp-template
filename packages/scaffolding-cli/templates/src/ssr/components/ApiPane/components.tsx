import styled from "@emotion/styled"
import {Box, Grid} from "@material-ui/core"
import theme from "../../config/theme"

const themeConstants = {
    gutter: "1rem",
    borderRadius: "0.5rem",
    darkestColor: "#212121",
    amidoColor: "#fecb07",
    minimumPadding: "0.5rem",
}

export const Container = styled(Grid)`
    padding-bottom: ${themeConstants.gutter};
`

export const Pane = styled(Box)`
    box-sizing: border-box;
    width: 100%;
    background-color: #fff;
    opacity: 0.9;
    margin-right: ${themeConstants.gutter};
    border-radius: ${themeConstants.borderRadius};
    -webkit-box-shadow: 5px 5px 16px -5px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 5px 5px 16px -5px rgba(0, 0, 0, 0.75);
    box-shadow: 5px 5px 16px -5px rgba(0, 0, 0, 0.75);
    min-height: 3rem;
    display: flex;
    justify-content: space-between;
    margin-top: ${themeConstants.minimumPadding};
`
