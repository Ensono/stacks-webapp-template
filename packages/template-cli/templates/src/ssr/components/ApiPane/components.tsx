import styled from "@emotion/styled";
import { Box, Grid } from "@material-ui/core";
import theme from "../../config/theme";

export const Container = styled(Grid)`
  padding-bottom: ${theme.gutter};
`;

export const Pane = styled(Box)`
  box-sizing: border-box;
  width: 100%;
  background-color: #fff;
  opacity: 0.9;
  margin-right: ${theme.gutter};
  border-radius: ${theme.borderRadius};
  -webkit-box-shadow: 5px 5px 16px -5px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 5px 5px 16px -5px rgba(0, 0, 0, 0.75);
  box-shadow: 5px 5px 16px -5px rgba(0, 0, 0, 0.75);
  min-height: 3rem;
  display: flex;
  justify-content: space-between;
  margin-top: ${theme.minimumPadding};
`;
