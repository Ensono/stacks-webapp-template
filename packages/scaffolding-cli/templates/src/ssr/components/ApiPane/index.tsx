import {Typography, CircularProgress} from "@material-ui/core"
import {RestaurantItemProps} from "interfaces/restaurant.interface"
import React from "react"
import RestaurantListComponent from "../RestaurantList"
import {Container} from "./components"

type ApiPaneProps = {
    menuItems: RestaurantItemProps[]
    isLoading: boolean
}

export const ApiPane = ({menuItems, isLoading}: ApiPaneProps) => {
    return (
        <Container item>
            <Typography variant="h1">Latest menus:</Typography>
            {isLoading && <CircularProgress size={26} />}
            {!isLoading && menuItems?.length == 0 && (
                <Typography variant="h2">No results</Typography>
            )}
            {!isLoading && menuItems?.length >= 1 && (
                <RestaurantListComponent restaurantList={menuItems} />
            )}
        </Container>
    )
}
