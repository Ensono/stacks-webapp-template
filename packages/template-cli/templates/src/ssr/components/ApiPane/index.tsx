import React, { useState } from "react";
import { Container, Pane } from "./components";
import { Button } from "@material-ui/core";
import api from "constants/apis/menu";
import axios from "axios";

type ApiPaneProps = {};

const ApiPane = (props: ApiPaneProps) => {
  const myapi = api("getMenuList");
  const [state, setState] = useState(null);
  const callApi = async () => {
    const res = await axios({
      method: myapi.method,
      url: myapi.internalEndpoint(10)
    });
    setState(res.data.results);
  };
  return (
    <Container item>
      <Pane>
        <h1>Get Menu List</h1>
        <Button onClick={callApi}>{myapi.method}</Button>
      </Pane>
      {state && (
        <Pane>
          <ul>
            {state.map(item => (
              <li>{item.name}</li>
            ))}
          </ul>
        </Pane>
      )}
    </Container>
  );
};

export default ApiPane;
