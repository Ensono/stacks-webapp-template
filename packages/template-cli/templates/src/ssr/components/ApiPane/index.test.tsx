import { fireEvent, render, screen, waitForElement } from "@testing-library/react"
import * as axios from 'axios'
import * as React from 'react'
import { toMatchDiffSnapshot } from 'snapshot-diff'
import ApiPane from '.'

jest.mock('axios', () =>
  jest.fn(() =>
    Promise.resolve({
      data: {
        results:
          [
            {
              id: 'e98583ad-0feb-4e48-9d4f-b20b09cb2633',
              name: 'Breakfast Menu',
              description: 'Eggs, Bread, Coffee and more',
            },
          ],
      },
    }),
  ),
)
const mockedAxios = axios as jest.Mocked<typeof axios>

test('renders ApiPane snapshot', () => {
  const {asFragment} = render(<ApiPane />)
  expect(asFragment()).toMatchSnapshot()
})

test('fires api call on button click', async () => {
  const {getByText,asFragment} = render(<ApiPane />)
  const firstRender = asFragment()
  
  fireEvent.click(screen.getByTestId('apiPaneBtn'))

  
  await waitForElement(() => screen.getByText('Breakfast Menu'))
  
  const resultNode = await screen.findByTestId('results')
  expect(resultNode).toHaveTextContent('Breakfast Menu')
  expect(mockedAxios).toHaveBeenCalledTimes(1)
  
  // To match re-rendered DOM snapshot after the api response
  expect.extend({toMatchDiffSnapshot})
  expect(firstRender).toMatchDiffSnapshot(asFragment())
})
