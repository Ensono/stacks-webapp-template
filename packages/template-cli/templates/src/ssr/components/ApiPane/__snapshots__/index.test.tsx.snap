// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ApiPane snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root css-1hq8d8z MuiGrid-item"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1"
    >
      Latest menus:
    </h1>
    0
  </div>
</DocumentFragment>
`;
